scripts/analyse-go-binary.py

```diff
@@ -0,0 +1,200 @@
+
+            off, size = find_longest_string(section_data)
+            # print(hex(pe.OPTIONAL_HEADER.ImageBase + off + section_va), hex(size))
+            final_size = size
+
+        if section_name == ".text":
+            section_va = section.VirtualAddress
+            section_size = section.SizeOfRawData
+            section_data = section.get_data(section_va, section_size)
+
+            strings = re.findall(combinedregex, section_data)
+
+            for string_tuple in strings:
+                for string in string_tuple:
+                    if string != b"":
+                        try:
+                            decoded_string = string.decode("utf-8")
+                            if decoded_string.isprintable() and len(string) >= min_length:
+                                addr = 0
+                                yield StaticString(string=string, offset=addr, encoding=StringEncoding.ASCII)
+                        except UnicodeDecodeError:
+                            pass
+
+        if section_name in (".rdata", ".data"):
+            section_va = section.VirtualAddress
+            section_size = section.SizeOfRawData
+            section_data = section.get_data(section_va, section_size)
+
+            for i in tqdm(range(0, len(section_data) - alignment // 2, alignment // 2)):
+                try:
+                    curr = section_data[i : i + alignment]
+                    s_off, s_size = struct.unpack(fmt, curr)
+
+                    if not s_off and not (1 <= s_size < 128):
+                        continue
+
+                    s_rva = s_off - pe.OPTIONAL_HEADER.ImageBase
+
+                    if not pe.get_section_by_rva(s_rva):
+                        continue
+
+                    addr = pe.OPTIONAL_HEADER.ImageBase + section_va + i
+
+                    try:
+                        string = pe.get_string_at_rva(s_rva, s_size).decode("ascii")
+                    except UnicodeDecodeError:
+                        continue
+
+                    if len(string) >= min_length and len(string) == s_size:
+                        yield StaticString(string=string, offset=addr, encoding=StringEncoding.ASCII)
+                except Exception as e:
+                    logger.error(f"Error: {e}")
+                    raise
+
+
+def main(argv=None):
+    parser = argparse.ArgumentParser(description="Get Go strings")
+    parser.add_argument("path", help="file or path to analyze")
+    parser.add_argument(
+        "-n",
+        "--minimum-length",
+        dest="min_length",
+        type=int,
+        default=MIN_STR_LEN,
+        help="minimum string length",
+    )
+    args = parser.parse_args(args=argv)
+
+    static_strings = extract_go_strings(args.path, min_length=args.min_length)
+
+    final_length = 0
+
+    for strings_obj in static_strings:
+        string = strings_obj.string
+        final_length += len(string)
+
+    try:
+        print(final_length * 100 / final_size)
+    except:
+        pass
```

Comment:
why would this throw an exception, let's handle it differently if final_size is 0, for example